internal/services/communication/email_service_domain_resource_test.go

```diff
@@ -0,0 +1,194 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package communication_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/communication/2023-03-31/domains"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance/check"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type EmailServiceDomainTestResource struct{}
+
+func TestAccEmailServiceDomain_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_email_communication_service_domain", "test")
+	r := EmailServiceDomainTestResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("from_sender_domain").IsNotEmpty(),
+				check.That(data.ResourceName).Key("mail_from_sender_domain").IsNotEmpty(),
+				check.That(data.ResourceName).Key("verification_records.#").HasValue("1"),
```

Comment:
These should be covered by the `data.ImportStep()` stage and should not need to be explicitly checked.
```suggestion
```